rustls/tests/api.rs

```diff
@@ -272,6 +272,36 @@ fn config_builder_for_client_rejects_incompatible_cipher_suites() {
     );
 }
 
+#[cfg(feature = "tls12")]
+#[test]
+fn config_builder_for_client_rejects_cipher_suites_without_compatible_kx_groups() {
```

Comment:
This looks good.

But I am considering that we may also need to add tests to cover the Server Behavior mentioned in RFC: https://datatracker.ietf.org/doc/html/rfc7919#section-4. This is also related to my comments at `rustls/src/server/tls12.rs`.

Mainly these paragraphes (1st & last two):

```text
   If a compatible TLS server receives a Supported Groups extension from
   a client that includes any FFDHE group (i.e., any codepoint between
   256 and 511, inclusive, even if unknown to the server), and if none
   of the client-proposed FFDHE groups are known and acceptable to the
   server, then the server MUST NOT select an FFDHE cipher suite.  In
   this case, the server SHOULD select an acceptable non-FFDHE cipher
   suite from the client's offered list.  If the extension is present
   with FFDHE groups, none of the client's offered groups are acceptable
   by the server, and none of the client's proposed non-FFDHE cipher
   suites are acceptable to the server, the server MUST end the
   connection with a fatal TLS alert of type insufficient_security(71).

   If a non-anonymous FFDHE cipher suite is selected and the TLS client
   has used this extension to offer an FFDHE group of comparable or
   greater strength than the server's public key, the server SHOULD
   select an FFDHE group at least as strong as the server's public key.
   For example, if the server has a 3072-bit RSA key and the client
   offers only ffdhe2048 and ffdhe4096, the server SHOULD select
   ffdhe4096.

   When an FFDHE cipher suite is selected and the client sends a
   ClientKeyExchange, the server MUST verify that 1 < dh_Yc < dh_p - 1.
   If dh_Yc is out of range, the server MUST terminate the connection
   with a fatal handshake_failure(40) alert.
```

Reply:
Good suggestion. Will do.

Reply:
Done: `fn test_server_picks_dhe_group_when_clienthello_has_no_dhe_group_in_groups_ext()`